framework/tester.py

```diff
@@ -77,7 +123,16 @@ def __create_client(self, client):
         populate_properties(client)
         cid = client['id']
         if client['type'] == 'deproxy':
-            self.__clients[cid] = self.__create_client_deproxy(client)
+            ip = None
+            if client.get('interface', False):
+                if not remote.wait_available():
+                    raise Exception("Client node is unavaliable")
```

Comment:
Normally we use `remote.wait_available()` only before the test, to wait for reboot of the remote node. This situation is little bit different. If I got it right, you care about situations when the remote node becomes unavailable due to network misconfiguration. In this case we will wait for it in current test for some time, but it'll never never go up again. And Do we need to check all nodes? Not only client?

One of the best practices in configuring networks remotely is to arm reboot timer. If connection is lost and remote host is not reachable any more, it will be rebooted back to original configuration. If everything is good, administrator is able to connect and disarm the timer. But I don't think that we need that practice in the test framework. Please just make the exception  message more accurate to show real issue here.

Reply:
The check needs to be done once before creating all clients.